src/ILICheck.Web/Extensions.cs

```diff
@@ -209,6 +212,65 @@ public static string GetLogFile(this IFileProvider fileProvider, LogType logType
             }
         }
 
+        /// <summary>
+        /// Gets the full command which can be executed on a shell in an Unix environment to validate
+        /// the specified <paramref name="transferFile"/> with ilivalidator including environment specific
+        /// command prefixes if applicable. The path created by this function uses the Unix path format.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="homeDirectory">The home directory path using the Unix path format.</param>
+        /// <param name="transferFile">The transfer file name.</param>
+        /// <param name="gpkgModelNames">The extracted model names. Optional.</param>
+        /// <returns>The ilivalidation command.</returns>
+        public static string GetIlivalidatorCommand(this IConfiguration configuration, string homeDirectory, string transferFile, string gpkgModelNames = null)
+        {
+            homeDirectory = Path.TrimEndingDirectorySeparator(homeDirectory);
+            var transferFileNameWithoutExtension = Path.GetFileNameWithoutExtension(transferFile);
+
+            var logPath = $"{homeDirectory}/{transferFileNameWithoutExtension}_log.log";
+            var xtfLogPath = $"{homeDirectory}/{transferFileNameWithoutExtension}_log.xtf";
+            var transferFilePath = $"{homeDirectory}/{transferFile}";
+            var commandPrefix = configuration.GetSection("Validation")["CommandPrefix"];
+            var options = $"--log {logPath} --xtflog {xtfLogPath}";
+            if (!string.IsNullOrEmpty(gpkgModelNames)) options = $"{options} --models \"{gpkgModelNames}\"";
+
+            return $"{commandPrefix} ilivalidator {options} \"{transferFilePath}\"".Trim();
+        }
+
+        /// <summary>
+        /// Asynchronously executes the given <paramref name="command"/> on the shell specified in <see cref="GetShellExecutable(IConfiguration)"/>.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="command">The command to execute.</param>
+        /// <param name="cancellationToken">An optional token to cancel the asynchronous operation.</param>
+        /// <returns>The exit code that the associated process specified when it terminated.</returns>
+        public static async Task<int> ExecuteCommandAsync(this IConfiguration configuration, string command, CancellationToken cancellationToken = default)
```

Comment:
PP: Diese beiden Extensions passen nicht so gut auf `IConfiguration`, da sie mit der Konfigurationsdatei relativ wenig zu tun haben. Die Methoden benötigen nur eine Einstellung aus der Konfiguration, die man von aussen auch hereingeben könnte (betrifft vermutlich auch andere Extensions in dieser Klasse) — thematisch haben sie aber sonst nichts mit `IConfiguration` zu tun.
Eine Alternative wären z.B. statische Methoden in einer ilivalidator-Hilfsklasse (für vereinfachtes Testing) oder direkt in der `Validator`-Klasse, da sie vermutlich auch nur von diesem aufgerufen werden.